src/sailfishapp_priv.cpp

```diff
@@ -95,9 +95,24 @@ configureApp(QGuiApplication *app)
     //  - Qt Quick Local Storage (QQmlEngine::offlineStoragePath())
 
     QString name = appName();
-    app->setOrganizationName(name);
-    app->setOrganizationDomain(name);
-    app->setApplicationName(name);
+    QString applicationName;
+    QString organizationName;
+    QString organizationDomain;
+
+    MDesktopEntry entry("/usr/share/applications/" + name + ".desktop");
+    if (entry.isSandboxed()) {
+        const auto section = QStringLiteral("X-Sailjail");
+        organizationName = entry.value(section, "OrganizationName");
+        organizationDomain = organizationName;
+        applicationName = entry.value(section, "ApplicationName");
+    }
+
+    app->setOrganizationName(
+            organizationName.isEmpty() ? name : organizationName);
+    app->setOrganizationDomain(
+            organizationDomain.isEmpty() ? name : organizationDomain);
+    app->setApplicationName(
+            applicationName.isEmpty() ? name : applicationName);
```

Comment:
I would move all of these inside the if (entry.isSandboxed()) - if the app is not sandboxed, these are going to be empty anyway. And since we are anyway reusing the OrganizationName for OrganizationDomain, I would use a simple if instead of the ternary operator.

Reply:
Do you think it's better now?